VideoLocker/src/androidTest/java/org/edx/mobile/test/NotificationTest.java

```diff
@@ -0,0 +1,118 @@
+    private final String  channelId2 = "channel_id_2";
+    private final String  channelId3 = "channel_id_3";
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        initDelegation();
+    }
+
+    private void initDelegation() {
+
+        if (manager == null) {
+            tracker = Mockito.mock(NotificationDelegate.class);
+
+            manager =   UserNotificationManager.getInstance();
+            // use mocked tracker
+            manager.setDelegate(tracker);
+        }
+    }
+
+
+    public void testNotificationPreference() throws Exception {
+
+        EdxLocalParseChannel channel1 =
+                new EdxLocalParseChannel( courseId1, channelId1, true);
+        EdxLocalParseChannel channel2 =
+                new EdxLocalParseChannel( courseId2, channelId2, false);
+
+        NotificationPreference pref = new NotificationPreference();
+        pref.add( channel1 ) ;
+        pref.add( channel2 ) ;
+
+        assertTrue("getByChannelId", pref.getByChannelId(channelId1).getChannelId().equals(channelId1));
+        assertTrue("getByCourseId", pref.getByCourseId(courseId1).getCourseId().equals(courseId1));
+
+        assertTrue( pref.size() == 2 );
+
+
+        CourseEntry courseEntry1 = new CourseEntry();
+        courseEntry1.setId( courseId1 );
+        CourseEntry courseEntry2 = new CourseEntry();
+        courseEntry2.setId( courseId2 );
+        CourseEntry courseEntry3 = new CourseEntry();
+        courseEntry3.setId( courseId3 );
+
+        List<CourseEntry> courseEntryList = new ArrayList<>();
+        courseEntryList.add(courseEntry1);
+        courseEntryList.add(courseEntry3);
+
+        // find a list of courseId which is not in the dictionary
+        List<CourseEntry> newCourses = pref.filterForNewCourses(courseEntryList);
+
+        assertTrue("filterForNewCourses", newCourses.size() == 1 && newCourses.get(0).getId().equals(courseId3));
+
+        //find all the course entry in saved preference which are not in the current active
+        courseEntryList.clear();
+        courseEntryList.add(courseEntry2);
+        courseEntryList.add(courseEntry3);
+        List<EdxLocalParseChannel> inactiveCourses = pref.filterForInactiveCourses(courseEntryList);
+
+
+        assertTrue("filterForInactiveCourses", newCourses.size() == 1 && newCourses.get(0).getId().equals(courseId2));
+
+        pref.removeByChannelId(channelId3);
+        assertTrue("removeByChannelId", pref.size() == 2);
+        pref.removeByCourseId(courseId3);
+        assertTrue( "removeByCourseId", pref.size() == 2 );
+        pref.removeByChannelId(channelId1);
+        assertTrue( "removeByChannelId", pref.size() == 1 );
+        pref.removeByCourseId(courseId2);
+        assertTrue( "removeByCourseId", pref.size() ==  0 );
+    }
+
+    /**
+     * nothing interesting here.
+     * @throws Exception
+     */
+    public void testNotificationManager() throws Exception {
```

Comment:
@aleffert  just make sure all the api calls for Notification Manager do not throw exception.